### DISCUSSION TITLE
Fill Vertical Gaps

### DISCUSSION THREAD

**Discussion by Kazagan** [QuestPDF/QuestPDF#606] [Q&A]
The example provided will be using inline, but I'm looking to have elements fill in vertical gaps using any layout style. 

In the image below, I'm looking to have the top of boxes snap to the bottom of boxes that don't fill the full area given to it. 
[ColorfulPdf.pdf](https://github.com/QuestPDF/QuestPDF/files/12030543/ColorfulPdf.pdf)

In a table style, the boxes would be of the same size, but once filled with data, there would be unnecessary space taken up by the cell, that I could l would like to have filled by the below cell.  Dynamically resizing cells 
[ColorFulTable.pdf](https://github.com/QuestPDF/QuestPDF/files/12030639/ColorFulTable.pdf)

Is there any way to do this?

**Comment by MarcinZiabek** [comment]
Hi 😁 Do I understand correctly that you are looking for the "masonry" layout? https://css-irl.info/masonry-in-css/

If you don't require your items to support paging, you can use the DynamicComponent to calculate the exact size and position of all elements.

**Reply by Kazagan** [reply]
Yes, I don't do design work much, so I wasn't aware of a specific term for it, but I am looking for a sort of masonry layout. Unfortunately it does need to support paging so I can't use the dynamic component.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
